examples/apps/simple_wallet/workload/simple_wallet_execute_io.cpp

```diff
@@ -0,0 +1,215 @@
+* limitations under the License.
+*/
+
+#include <string>
+#include <string.h>
+
+#include "types.h"
+#include "jsonvalue.h"
+#include "json_utils.h"
+#include "crypto_utils.h"
+#include "error.h"
+#include "tcf_error.h"
+#include "enclave_utils.h"
+#include <parson.h>
+
+#include "simple_wallet_execute_io.h"
+#include "file_io.h"
+//#include "workload_crypto_utils.h"
+
+#define LEDGER_FILE "/tmp/account_ledger.json"
+
+SimpleWalletLedger* SimpleWalletLedger::instance = 0;
+
+SimpleWalletIoExecutor::SimpleWalletIoExecutor() {}
+
+// Create encryption key used for encrypting or decrypting
+// ledger having all wallet details
+bool SimpleWalletIoExecutor::CreateLedgerEncryptionKey() {
+    std::string enc_key = tcf::crypto::CreateHexEncodedEncryptionKey();
+    if (enc_key.empty()) {
+        return false;
+    }
+    SimpleWalletLedger* ledger = SimpleWalletLedger::GetInstance();
+    ledger->ledger_encryption_key = enc_key;
+    return true;
+}
+
+SimpleWalletLedger* SimpleWalletLedger::GetInstance() {
+    if(!SimpleWalletLedger::instance) {
+        SimpleWalletLedger::instance = new SimpleWalletLedger();
+    }
+    return SimpleWalletLedger::instance;
+}
+
+// Create Wallet ledger file to store details of transactions in encrypted format
+void SimpleWalletIoExecutor::CreateWalletLedger() {
+    uint32_t status = ExecuteFileWrite("");
+    if (status == 0) {
+        Log(TCF_LOG_INFO, "Successfully created wallet ledger file %s",
+            LEDGER_FILE);
+    } else {
+        Log(TCF_LOG_INFO, "Failed to create wallet ledger file %s",
+            LEDGER_FILE);
+    }
+}
+
+// Get wallet balance for given account from json serialized ledger
+std::string SimpleWalletIoExecutor::GetWalletBalanceFromLedger(std::string ledger_data,
+    std::string account_name) {
+    // Parse account_ledger.json data
+    JsonValue parsed(json_parse_string(ledger_data.c_str()));
+    tcf::error::ThrowIfNull(parsed.value,
+        "failed to parse the work order request, badly formed JSON");
+
+    JSON_Object* ledger_object = json_value_get_object(parsed);
+    tcf::error::ThrowIfNull(ledger_object, "Missing JSON object in ledger");
+    Log(TCF_LOG_INFO, "Getting balance for the account from ledger object: %s",
+        account_name);
+    std::string balance = GetJsonStr(ledger_object, account_name.c_str(),
+        nullptr);
+    Log(TCF_LOG_INFO, "GetWalletBalanceFromLedger::Got balance for account %s",
+        account_name);
+    return balance;
+}
+
+std::string SimpleWalletIoExecutor::UpdateLedger(std::string ledger_data,
+    std::string account_name, std::string amount) {
+    JSON_Value* ledger_val;
+    if (ledger_data.empty()) {
+        Log(TCF_LOG_INFO, "Creating first wallet in ledger");
+        ledger_val = json_value_init_object();
+    } else {
+        // Parse account_ledger.json data
+        ledger_val = json_parse_string(ledger_data.c_str());
+        tcf::error::ThrowIfNull(
```

Comment:
This function is throwing an exception on error but other functions returns empty string on error. is caller catching this exception?

Reply:
Ok, will handle the exceptions.